ml-agents/mlagents/trainers/trainer/trainer.py

```diff
@@ -116,13 +118,35 @@ def save_model(self, name_behavior_id: str) -> None:
         """
         self.get_policy(name_behavior_id).save_model(self.get_step)
 
-    def export_model(self, name_behavior_id: str) -> None:
+    def export_model(self, name_behavior_id: str, is_checkpoint: bool = False) -> None:
         """
         Exports the model
         """
         policy = self.get_policy(name_behavior_id)
-        settings = SerializationSettings(policy.model_path, policy.brain.brain_name)
-        export_policy_model(settings, policy.graph, policy.sess)
+        if is_checkpoint:
+            checkpoint_path = f"{name_behavior_id}-{self.get_step}"
+            settings = SerializationSettings(
+                policy.model_path, policy.brain.brain_name, checkpoint_path
+            )
+            new_checkpoint: Dict[str, Union[int, str]] = {}
+            # Store steps and file_path
+            new_checkpoint["steps"] = int(self.get_step)
+            new_checkpoint["file_path"] = os.path.join(
+                settings.model_path, f"{settings.checkpoint_path}.nn"
+            )
```

Comment:
I think this should be a dataclass, e.g.:
```python3
@dataclass 
class Checkpoint:
  steps: int
  file_path: str
```

Reply:
For consistency, maybe an `attrs` class?

```
@attr.s(auto_attribs=True)
class Checkpoint:
  steps: int
  file_path: str
```

Reply:
Thanks @ervteng -- my mistake 😓